adv-rumors.lua

```diff
@@ -68,6 +61,17 @@ function renameChoice(text, choice)
     end
 end
 
+function getKeywords(choice)
+    local keywords = {}
+    for i, keyword in ipairs(choice.key_word) do
+        local keytext = dfhack.df2utf(keyword.value):lower()
+        if not keywords[keytext] then
+            table.insert(keywords, keytext)
+        end
```

Comment:
`keywords[keytext]` will always be `nil` since `keywords` is a list, not a map. If you also want a map/set for detecting duplicates, you have to maintain that separately (like in the `local new_keywords, keywords_set` code below)

I'm not sure if you need to check for duplicates at all, though, since the list in `choice.key_word` *shouldn't* have any.